lib/serialize.js

```diff
@@ -70,11 +71,18 @@ module.exports = function (app, defaults) {
 
     var relation = model.relations[relationName];
     if (relationName && relation) {
-      relatedModelPlural = utils.pluralForModel(relation.modelTo);
-      primaryKeyField = utils.primaryKeyForModel(relation.modelTo);
+      if (relation.polymorphic && utils.relationFkOnModelFrom(relation)) {
+        var discriminator = utils.clone(ctx.instance)[relation.polymorphic.discriminator];
+        relatedModel = app.models[discriminator];
+      } else {
+        relatedModel = relation.modelTo;
+      }
+      relatedModelPlural = utils.pluralForModel(relatedModel);
+      primaryKeyField = utils.primaryKeyForModel(relatedModel);
 
       if (relatedModelPlural) {
         type = relatedModelPlural;
+        model = relatedModel;
```

Comment:
@Resonance1584 

model is used in the serializer to check for serialize and deserialize hooks `beforeJsonApiDeserialize` etc.

I think this change will cause some issues with that as suddenly the model we are checking for hooks changes. Trying to think this through.

GET /posts/1
We should definitely be calling `Post.jsonApiSerialize` if it exists

GET /posts/1/comments
Should we be calling  `Post.jsonApiSerialize` or  `Comment.jsonApiSerialize`?